yarn-project/cli/src/cmds/l1/index.ts

```diff
@@ -326,6 +327,46 @@ export function injectCommands(program: Command, log: LogFn, debugLogger: Logger
       });
     });
 
+  program
+    .command('add-l1-validator-via-rollup')
+    .description('Adds a validator to the L1 rollup contract via a direct deposit.')
+    .addOption(l1RpcUrlsOption)
+    .option('-pk, --private-key <string>', 'The private key to use sending the transaction', PRIVATE_KEY)
+    .option(
+      '-m, --mnemonic <string>',
+      'The mnemonic to use sending the transaction',
+      'test test test test test test test test test test test junk',
+    )
+    .addOption(l1ChainIdOption)
+    .option('--attester <address>', 'ethereum address of the attester', parseEthereumAddress)
+    .option('--withdrawer <address>', 'ethereum address of the withdrawer', parseEthereumAddress)
+    .option(
+      '--bls-secret-key <string>',
+      'The BN254 scalar field element used as a secret key for BLS signatures. Will be associated with the attester address.',
+      parseBigint,
+    )
+    .option('--moveWithLatestRollup', 'Whether to move with the latest rollup', true)
+    .option('--rollup <string>', 'Rollup contract address or network name')
```

Comment:
I propose making this only take an optional rollup address. The `--network` flag/`NETWORK` env var is already handled at the top of the aztec binary. We can just check if we have a `REGISTRY_CONTRACT_ADDRESS` env var (via enrichment) and take the canonical rollup from it.

Reply:
Yes, good idea.